### PR TITLE
KYLIN-4817 Refine CubeMigrationCLI for kylin4

### PR DESCRIPTION
**Pull request by zhangayqian**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [ ] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: kylin-spark-project/kylin-spark-common/src/main/java/org/apache/kylin/engine/spark/metadata/cube/PathManager.java [hunk 1]
```diff
@@ -49,6 +49,12 @@ public static String getSegmentParquetStoragePath(CubeInstance cube, String segN
         return hdfsWorkDir + "parquet" + File.separator + cube.getName() + File.separator + segName + "_" + identifier;
     }
 
+    public static String getSegmentParquetStoragePath(String hdfsWorkDir, String cubeName, CubeSegment segment) {
+        String segmentName = segment.getName();
+        String identifier = segment.getStorageLocationIdentifier();
+        return hdfsWorkDir + "parquet" + File.separator + cubeName + File.separator + segmentName + "_" + identifier;
```
Comment (hit-lacus) [inline kylin-spark-project/kylin-spark-common/src/main/java/org/apache/kylin/engine/spark/metadata/cube/PathManager.java:55]: Is there a `File.separator` between `hdfsWorkDir` and `parquet`

**Comment by zhangayqian** [comment]
### Test Evidence
- src kylin instance url: ADMIN:KYLIN@10.1.3.94:7070
- target Kylin instance url: ADMIN:KYLIN@10.1.3.93:7072
- migrate cube: kylin_sales_cube
- src project: learn_kylin
- target project: test_migration_kylin

####
- cmd
```
bin/kylin.sh org.apache.kylin.tool.CubeMigrationCLI ADMIN:KYLIN@10.1.3.94:7070 ADMIN:KYLIN@10.1.3.93:7072 kylin_sales_cube test_migration_kylin true false true true true
```
- Result: `kylin_sales_cube` be migrated to project `test_migration_kylin` of 10.1.3.93:7072, and the related hdfs resource(dict、snapshot、parquet file) be copied to target kylin instance HDFS working dir.
- Migrate log
image
- Migrated can be queried
image

File: kylin-spark-project/kylin-spark-common/src/main/java/org/apache/kylin/engine/spark/metadata/cube/PathManager.java (hunk 1 above)
Comment (zhangayqian) [reply kylin-spark-project/kylin-spark-common/src/main/java/org/apache/kylin/engine/spark/metadata/cube/PathManager.java:55]: Actually  the end of `hdfsWorkDir` already comes with a File.separator.

Comment (hit-lacus) [reply kylin-spark-project/kylin-spark-common/src/main/java/org/apache/kylin/engine/spark/metadata/cube/PathManager.java:55]: OK

File: tool/src/main/java/org/apache/kylin/tool/CubeMigrationCLI.java [hunk 2]
```diff
@@ -192,20 +184,12 @@ public void moveCube(KylinConfig srcCfg, KylinConfig dstCfg, String cubeName, St
...
-                checkMigrationSuccess(dstConfig, cubeName, true);
-            }
             updateMeta(dstConfig, projectName, cubeName, cube.getModel());
         } else {
             showOpts();
         }
     }
 
-    public void checkMigrationSuccess(KylinConfig kylinConfig, String cubeName, Boolean ifFix) throws IOException {
-        CubeMigrationCheckCLI checkCLI = new CubeMigrationCheckCLI(kylinConfig, ifFix);
```
Comment (hit-lacus) [inline tool/src/main/java/org/apache/kylin/tool/CubeMigrationCLI.java:205]: After remove HBase, looks like we do not need `CubeMigrationCheckCLI` any more.

File: tool/src/main/java/org/apache/kylin/tool/CubeMigrationCLI.java [hunk 3]
```diff
@@ -326,22 +292,32 @@ protected void listCubeRelatedResources(CubeInstance cube, List<String> metaReso
         metaResource.addAll(getCompatibleTablePath(tblRefs, prj, ResourceStore.TABLE_EXD_RESOURCE_ROOT));
 
         if (doMigrateSegment) {
+            for (DictionaryDesc dictionaryDesc : cubeDesc.getDictionaries()) {
+                String[] columnInfo = dictionaryDesc.getColumnRef().getColumnWithTable().split("\\.");
+                String globalDictPath;
+                if (columnInfo.length == 3) {
```
Comment (hit-lacus) [inline tool/src/main/java/org/apache/kylin/tool/CubeMigrationCLI.java:298]: Magic number, isn't it?

Comment (hit-lacus) [inline tool/src/main/java/org/apache/kylin/tool/CubeMigrationCLI.java:298]: In which case `columnInfo.length` is not 3?

**Review by hit-lacus — APPROVED** [review]
Approved.

Thank you for contribution. @zhangayqian

**Pull request merged**